platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -323,14 +352,76 @@ - (MGLImage *)blurredAttributionBackground:(CIImage *)backgroundImage
     CGImageRef cgimg = [ctx createCGImage:blurredImage fromRect:[backgroundImage extent]];
     
 #if TARGET_OS_IPHONE
-    
     return [UIImage imageWithCGImage:cgimg];
 #else
-    
     return [[NSImage alloc] initWithCGImage:cgimg size:[backgroundImage extent].size];
 #endif
 }
 
+- (MGLSnapshotAttributionOptions *)attributionOptionsForSize:(CGSize)snapshotSize attributionInfo:(NSArray *)attributionInfo
+{
+    NSMutableArray *attributionOptions = [NSMutableArray array];
+    MGLSnapshotAttributionOptions *largeLogoAttribution = [self attributionOptionsForAttributionInfo:attributionInfo abbreviated:NO];
+#if TARGET_OS_IPHONE
+    largeLogoAttribution.logoImage = [UIImage imageNamed:@"mapbox" inBundle:[NSBundle mgl_frameworkBundle] compatibleWithTraitCollection:nil];
+#else
+    largeLogoAttribution.logoImage = [[NSImage alloc] initWithContentsOfFile:[[NSBundle mgl_frameworkBundle] pathForResource:@"mapbox" ofType:@"pdf"]];
+#endif
+    _defaultLogoHeight = largeLogoAttribution.logoImage.size.height;
+    
+    MGLSnapshotAttributionOptions *smallLogoAttribution = [self attributionOptionsForAttributionInfo:attributionInfo abbreviated:NO];
+#if TARGET_OS_IPHONE
+    smallLogoAttribution.logoImage = [UIImage imageNamed:@"mapbox_helmet" inBundle:[NSBundle mgl_frameworkBundle] compatibleWithTraitCollection:nil];
+#else
+    smallLogoAttribution.logoImage = [[NSImage alloc] initWithContentsOfFile:[[NSBundle mgl_frameworkBundle] pathForResource:@"mapbox_helmet" ofType:@"pdf"]];
+#endif
+    
+    MGLSnapshotAttributionOptions *noLogoAttribution = [self attributionOptionsForAttributionInfo:attributionInfo abbreviated:YES];
+    
+    [attributionOptions addObject:largeLogoAttribution];
+    [attributionOptions addObject:smallLogoAttribution];
+    [attributionOptions addObject:noLogoAttribution];
+    
+    for (MGLSnapshotAttributionOptions *attributionOption in attributionOptions) {
+        // -[Mapbox Logo]-[Attribution Background]-
+        CGFloat origin = attributionOption.logoImage ? MGLLogoImagePosition.x : 0;
+        CGFloat width = origin + attributionOption.logoImage.size.width + 10 + attributionOption.attributionBackgroundSize.width + 10;
+        if (width <= snapshotSize.width) {
+            return attributionOption;
+        }
+    }
+    
+    return noLogoAttribution;
+}
+
+- (MGLSnapshotAttributionOptions *)attributionOptionsForAttributionInfo:(NSArray *)attributionInfo abbreviated:(BOOL)abbreviated
+{
+    NSString *openStreetMap = @"OpenStreetMap";
+    NSString *OSM = @"OSM";
```

Comment:
Make these strings localizable, in case one of the localizations translates “OpenStreetMap” and “OSM” into a different script (as Chinese and Japanese probably should).